modules/ghlint-api/src/main/kotlin/net/twisterrob/ghlint/model/Permission.kt

```diff
@@ -0,0 +1,99 @@
+package net.twisterrob.ghlint.model
+
+public sealed interface Scope {
```

Comment:
Could be an enum, or a value class. (Scope doesn't **have** an `access`, it's just "what" a token enables access to, right?)

Reply:
Technically a Scope fully encapsulates everything - it represents everything a system needs to know, e.g.,

`packages:read` is one scope, `packages:write` is another. So I'd say it does have an access